viewing_party/party.py

```diff
@@ -1,23 +1,160 @@
-# ------------- WAVE 1 --------------------
+import statistics
+from statistics import mode
 
+# -----------------------------------------
+# ------------- WAVE 1 --------------------
+# -----------------------------------------
 def create_movie(title, genre, rating):
-    pass
+    """Creates a movie dict."""
+    if not title or not genre or not rating:
+        return None
+    return {
+        "title": title,
+        "genre": genre,
+        "rating": rating
+    }
+
+def add_to_watched(user_data, movie):
+    """Adds movie to a user's watched list."""
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    """Adds a movie to a user's watchlist."""
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, movie):
+    """Moves a movie from a user's watch list to their watched list."""
+    for i in range(len(user_data["watchlist"])):
+        if user_data["watchlist"][i]["title"] == movie:
+            movie_dict = user_data["watchlist"].pop(i)
+            user_data["watched"].append(movie_dict)

```

Comment:
While this 💯 works, we can also make our code more readable by iterating our for loop via a `<x> in <iterable_type>` syntax, instead of `i in range(len(<iterable_type>))` like so:
```
for user_movie in user_data["watchlist"]:
    if user_movie["title"] == movie:
        movie_dict = user_data["watchlist"].pop(user_movie)
        ...
```